### PR TITLE
Add `prompts_and_responses` support to `invoke_and_assert`

### PR DESCRIPTION
**Pull request by bunchesofdonald**
This adds a new concept of `prompts_and_responses` to `invoke_and_assert`. `prompts_and_responses` let's you define the prompts you expect and the response that should be given, and includes support for navigating tables with up/down arrows. `invoke_and_assert` will grab all of the 'responses' and use those as input for the command, and then it will assert that all of the prompts are in the output, and in the expected order. It will also assert that the correct option is selected in the table. 

I've updated several tests with this new pattern, but wanted to get feedback before going further.

### CODE REVIEW

**Review by chrisguidry — APPROVED** [review]
That is really cool!

File: tests/cli/test_deploy.py [hunk 1]
```diff
@@ -4807,36 +4788,19 @@ def test_save_user_inputs_with_interval_schedule(self):
...
-                + "inflatable"
-                + readchar.key.ENTER
-                # accept save user inputs
-                + "y"
-                + readchar.key.ENTER
-            ),
+            prompts_and_responses=[
+                ("? Deployment name (default)", ""),
+                ("Would you like to configure schedules for this deployment?", ""),
+                ("What type of schedule would you like to use?", "↓", "Cron"),
```
Comment (bunchesofdonald) [inline tests/cli/test_deploy.py:4794]: This is an example of using the down arrow (`↓`) to select an option in the table. You can also use `readchar.key.DOWN` here if you prefer.

File: tests/cli/test_deploy.py [hunk 2]
```diff
@@ -4929,18 +4872,7 @@ def test_deploy_existing_deployment_with_no_changes_does_not_prompt_save(self):
...
-                + readchar.key.ENTER
-                # accept create work pool
-                + readchar.key.ENTER
-                # choose process work pool
-                + readchar.key.ENTER
-                # enter work pool name
-                + "inflatable"
-                + readchar.key.ENTER
-            ),
+            prompts_and_responses=[("Would you like to save configuration", "y")],
```
Comment (bunchesofdonald) [inline tests/cli/test_deploy.py:4875]: This is an example of what i'd consider a false negative, and one of the issues with our current pattern. This command didn't actually result in asking the user about remote storage or work pools. Previously the input was just ignored, but with the `prompts_and_responses` the test would have failed.

**Comment by bunchesofdonald** [comment]
@serinamarie and @desertaxle wanted to make sure you both had a look at this before I merged it, since this is kinda your domain.

**Review by desertaxle — APPROVED** [review]
SLICK

**Review by serinamarie — APPROVED** [review]
Amazing!!

**Pull request merged**